pkg/webhook/mutatingwebhook.go

```diff
@@ -174,3 +191,53 @@ func parseSidecarContainerImage(pod *corev1.Pod) (string, error) {
 
 	return image, nil
 }
+
+func (si *SidecarInjector) supportsNativeSidecar() (bool, error) {
+	list, err := si.NodeLister.List(labels.Everything())
+	if err != nil {
+		return false, fmt.Errorf("failed to get cluster nodes: %w", err)
```

Comment:
## Logging vs returning error
This failure seems important enough to reject admission of pod. However, an alternative is to log an error and use regular sidecar.